zulipterminal/model.py

```diff
@@ -945,9 +945,20 @@ def _handle_update_message_event(self, event: Event) -> None:
             # the event didn't have a 'subject' update.
             if 'subject' in event:
                 new_subject = event['subject']
+                stream_id = event['stream_id']
                 for msg_id in event['message_ids']:
                     self.index['messages'][msg_id]['subject'] = new_subject
                     self._update_rendered_view(msg_id)
+                if stream_id in self.index['topics']:
+                    # Reload topic list.
+                    self._fetch_topics_in_streams([stream_id])
```

Comment:
@neiljp No, as I mentioned on czo, due to the topic data in the index never being complete, there is no way of knowing when a topic is empty (i.e doesn't have any messages left).

It would be useful if the backend response could include an extra flag when this happens, though it's not present currently AFAIK.

Reply:
Let's discuss on #zulip-terminal.